### PR TITLE
SOV-1648: User's LOC is not highlighted in the chart

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-1648

### CODE REVIEW

File: apps/frontend/src/app/3_organisms/LOCChart/LOCChart.tsx [hunk 1]
```diff
@@ -125,17 +114,13 @@ export const LOCChart: FC<LOCChartProps> = ({ isDefaultView = false }) => {
...
-    refetch: refetchTroves,
-  } = useGetTroves();
+  const { data: troves, loading: loadingTroves } = useGetTroves();
 
   useEffect(() => {
     if (troves && !loadingTroves && globalsEntity) {
       // load and parse data and then show it immediately if wallet not connected
       // logic for connected wallets is in another useEffect
-      const updatedTroves = troves.troves.map((trove: TroveData) => {
+      const updatedTroves = troves.troves.map(trove => {
```
Comment (soulBit) [inline apps/frontend/src/app/3_organisms/LOCChart/LOCChart.tsx]:
```suggestion
      const updatedTroves = troves.troves?.map(trove => {
```

File: apps/frontend/src/app/3_organisms/LOCChart/hooks/useGetUserOpenTrove.ts [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-    () => ({
-      user: account,
-    }),
-    [account],
-  );
+  const { account } = useAccount();
 
-  return useQuery(GetUserOpenTroveDocument, {
-    variables: troveConfig,
+  const { data, loading } = useGetUserOpenTroveQuery({
```
Comment (soulBit) [inline apps/frontend/src/app/3_organisms/LOCChart/hooks/useGetUserOpenTrove.ts]: When testing I can see this issue that is currently working fine on `develop`:
https://sovryn.atlassian.net/browse/SOV-1880

Chart should refresh whenever trove state changes

File: apps/frontend/src/app/3_organisms/LOCChart/LOCChart.tsx [hunk 3]
```diff
@@ -165,12 +150,12 @@ export const LOCChart: FC<LOCChartProps> = ({ isDefaultView = false }) => {
             debtAmount,
             price,
           ),
-        };
+        } as TroveData;
```
Comment (soulBit) [inline apps/frontend/src/app/3_organisms/LOCChart/LOCChart.tsx]:
```suggestion
        return {
          id: trove.id,
          collateral: collateralAmount,
          debt: debtAmount,
          collateralRatioSortKey_legacy: trove.collateralRatioSortKey_legacy ?? '',
          collateralRatio: calculateCollateralRatio(
            collateralAmount,
            debtAmount,
            price,
          ),
        };
```

Potentially null/undefined value for `collateralRatioSortKey_legacy` is what causes the typecheck error, this code resolves it - unless you were worried about empty values?

**Review by soulBit — CHANGES_REQUESTED** [review]

**Review by soulBit — APPROVED** [review]

File: apps/frontend/src/app/3_organisms/LOCChart/LOCChart.tsx [hunk 4]
```diff
@@ -113,7 +107,7 @@ export const LOCChart: FC<LOCChartProps> = ({ isDefaultView = false }) => {
           backgroundColor: bar => {
             if (bar.raw) {
               const { id } = bar.raw as TroveData;
-              return id === account
+              return id.toLowerCase() === account.toLowerCase()
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/LOCChart/LOCChart.tsx]: this comparison is being used a lot in the dapp. please add a function in utils and use it here:
```
export const areAddressesEqual = (address1: string, address2: string) =>
  address1.toLowerCase() === address2.toLowerCase();
```

**Review by Rickk137 — APPROVED** [review]

**Pull request merged**